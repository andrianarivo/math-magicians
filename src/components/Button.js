import PropTypes from 'prop-types';
import styles from '../styles/Button.module.css';

function Button(props) {
  const { text, textAlign, flex, color, textColor, cursor, onClick, testId } =
    props;
  let alignment = 'center';
  let paddingRight = '0';
  switch (textAlign) {
    case 'right':
      alignment = 'flex-end';
      paddingRight = '2em';
      break;
    default:
    case 'center':
      break;
  }

  const handleClick = () => {
    onClick(text);
  };

  return (
    <button
      data-testid={testId}
      className={styles.button}
      style={{
        justifyContent: alignment,
        color: textColor,
        flex,
        backgroundColor: color,
        paddingRight,
        cursor,
      }}
      type="button"
      onClick={handleClick}
      onKeyDown={handleClick}
    >
      <p>{text}</p>
    </button>
  );
}

Button.defaultProps = {
  textAlign: 'center',
  flex: 1,
  color: '#e0e0e0',
  textColor: '#000',
  cursor: 'pointer',
  testId: 'button',
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  testId: PropTypes.string,
  textAlign: PropTypes.string,
  flex: PropTypes.number,
  color: PropTypes.string,
  textColor: PropTypes.string,
  cursor: PropTypes.string,
};

export default Button;
