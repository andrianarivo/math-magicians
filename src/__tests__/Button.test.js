import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from '../components/Button';

describe('testing Button component', () => {
  test('render', () => {
    const tree = renderer.create(<Button onClick={() => {}} text="1" />);
    expect(tree).toMatchSnapshot();
  });

  test('user interaction', () => {
    let value = 'fail';
    render(
      <Button
        onClick={() => {
          value = 'clicked';
        }}
        text="1"
      />
    );

    fireEvent.click(screen.getByText('1'));

    expect(value).toBe('clicked');
  });
});
