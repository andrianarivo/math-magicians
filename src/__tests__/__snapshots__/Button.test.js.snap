// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button component render 1`] = `
<button
  className="button"
  data-testid="button"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "backgroundColor": "#e0e0e0",
      "color": "#000",
      "cursor": "pointer",
      "flex": 1,
      "justifyContent": "center",
      "paddingRight": "0",
    }
  }
  type="button"
>
  <p>
    1
  </p>
</button>
`;
